### PR TITLE
Remove narrow import exception condition (#7259)

### PR DESCRIPTION
**Pull request by mathieuboudreau**
# References and relevant issues

#7259

# Description

It resolves a bug on where, at least on (my) MacM1, a different exception isn't caught when trying to import numba. This made it such that napari would crash when loading layers with a colormap.

- [x] My PR is the minimum possible work for the desired functionality

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to docstrings and documentation
  (open a PR on the docs repository (https://github.com/napari/docs) if relevant!)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans._("some string")` to make them localizable.
  (For more information see our [translations guide](https://napari.org/stable/developers/contributing/translations.html)).
-->

### CODE REVIEW

**Comment by melonora** [comment]
Hi @mathieuboudreau ,

Thanks for your contribution! I am not aware of the issue, but can check with a borrowed mac whether I can reproduce. Which python version are you running?

**Comment by mathieuboudreau** [comment]
(venv_ads) mathieuboudreau@Mathieus-MacBook-Pro axondeepseg % napari --info
napari: 0.5.3
Platform: macOS-14.6.1-x86_64-i386-64bit
System: MacOS 14.6.1
Python: 3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:45:13) [Clang 16.0.6 ]
Qt: 5.15.14
PyQt5: 5.15.11
NumPy: 1.26.4
SciPy: 1.14.1
Dask: 2024.8.1
VisPy: 0.14.3
magicgui: 0.9.1
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.8
npe2: 0.7.7

OpenGL:

GL version: 2.1 Metal - 88.1
MAX_TEXTURE_SIZE: 16384
GL_MAX_3D_TEXTURE_SIZE: 2048
Screens:

screen 1: resolution 1920x1080, scale 1.0
Optional:

numba: 0.60.0
triangle: 20230923
napari-plugin-manager: 0.1.0
Settings path:

/Users/mathieuboudreau/Library/Application Support/napari/venv_ads_514b6e6e5dfe9208f37d8f3677c5ee0ad1feec07/settings.yaml
Plugins:
napari: 0.5.3 (81 contributions)
napari-ADS: 1.0.0 (2 contributions)
napari-console: 0.0.9 (0 contributions)
napari-svg: 0.2.0 (2 contributions)

**Comment by mathieuboudreau** [comment]
> Hi @mathieuboudreau ,
> 
> Thanks for your contribution! I am not aware of the issue, but can check with a borrowed mac whether I can reproduce. Which python version are you running?

For me, without this fix, the following script would fail:

```python
import torch
import napari
import imageio.v3 as iio
import numpy as np

viewer = napari.Viewer()
mask = iio.imread("mask.png")  # Set the value of mask.png to the correct path
axon_data = (mask[:, :, 0] == 255).astype(np.uint8)
myelin_data = (mask[:, :, 0] == 128).astype(np.uint8)

viewer.add_labels(
    axon_data,
    colormap={ None: 'transparent', 1: "blue"},
    name="axon",
    metadata={"associated_image_name": "image"}
)

viewer.add_labels(
    myelin_data,
    colormap={ None: 'transparent', 1: "red"},
    name="myelin",
    metadata={"associated_image_name": "image"}
)

napari.run()
```

with the image in this comment: https://github.com/napari/napari/issues/7259#issuecomment-2344970738

File: napari/utils/colormaps/_accelerated_cmap.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def _labels_raw_to_texture_direct_inner_loop(
 
 try:
     import numba
-except ModuleNotFoundError:
+except:
```
Comment (Czaki) [inline napari/utils/colormaps/_accelerated_cmap.py:218]:
```suggestion
except ImportError:
```

This will be enough to handle any problem with import

**Comment by Czaki** [comment]
The general problem with this PR is that we need numba to be able to handle `DirectColormap` for values above 2**16. So, changing this exception may lead to confusion of users why napari does not work with their data even when they have `numba` installed.

**Comment by mathieuboudreau** [comment]
> The general problem with this PR is that we need numba to be able to handle `DirectColormap` for values above 2**16. So, changing this exception may lead to confusion of users why napari does not work with their data even when they have `numba` installed.

Ok, I'm open to other solutions to fix the bug that broke our Napari plugin - it's not quite clear why numba is required just to add a red or blue label, I can see some edge cases but at the moment it's holding us up in https://github.com/axondeepseg/axondeepseg/pull/800 to release v5. I guess maybe we could rollback napari to a fixed earlier version, but that would not be great in the long run. Alternatively, we could also fork napari and continue with our own edits on that version for our release, also not ideal.

I'll restart from a fresh install of everything tomorrow, as the `ImportError` solution isn't working for me now and I don't know why. Debugging https://github.com/napari/napari/issues/7259 required me to completely edit many files, let see if I can reproduce my fix in the meantime.

**Comment by Czaki** [comment]
> Ok, I'm open to other solutions to fix the bug that broke our Napari plugin - it's not quite clear why numba is required just to add a red or blue label

It is required to do it faster with using less memory.

**Comment by mathieuboudreau** [comment]
I just tested from fresh, and my original PR doesn't solve the bug, neither does using `ImportError` instead. The only thing that solves is is completely omitting the entire ``try`  and replacing that entire block:

https://github.com/napari/napari/blob/c9c5ad84fe131e7d4b1e61fcef25c63a3bc33314/napari/utils/colormaps/_accelerated_cmap.py#L216-L233

with only this:

https://github.com/napari/napari/blob/c9c5ad84fe131e7d4b1e61fcef25c63a3bc33314/napari/utils/colormaps/_accelerated_cmap.py#L219-L221

I screencaptured my entire setup and test / solution and will share it here after it's been processed, but this means this PR is no longer the proper solution, so I'm closing it.

**Comment by Czaki** [comment]
I just tested the provided script with a fresh environment on my macos (M2) machine, and it works without any error on main. 
Did you installed torch from pypi?

**Comment by mathieuboudreau** [comment]
> I just tested the provided script with a fresh environment on my macos (M2) machine, and it works without any error on main. Did you installed torch from pypi?

Here's a screencapture'd video of every step I did: https://www.youtube.com/watch?v=yVyYeDaQSes
